.pre-commit-config.yaml

```diff
@@ -0,0 +1,57 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-executables-have-shebangs
+    -   id: check-merge-conflict
+    -   id: debug-statements
+    -   id: check-yaml
+        files: .*\.(yaml|yml)$
+
+-   repo: https://github.com/asottile/reorder_python_imports
+    rev: v2.6.0
+    hooks:
+    -   id: reorder-python-imports
+        files: ^src/|tests/
+
+-   repo: https://github.com/pre-commit/mirrors-mypy.git
+    rev: v0.910
+    hooks:
+    -   id: mypy
+        files: ^src/|tests/
+        # 😔 These `additional_dependencies` mirror those in the `dev` extra of `setup.cfg` and are needed
+        # to run `mypy` under `tox`. This violates "DRY" but we don't have a better way of doing this—yet.
+        additional_dependencies: [
+            lxml-stubs,
+            pandas-stubs,
+            types-flask,
+            types-jsonschema,
+            types-pkg_resources,
+            types-python-dateutil,
+            types-requests,
+            types-six,
+            types-waitress
+        ]
```

Comment:
I have a fix for this duplication going up in the template repo very soon

Reply:
See [this PR](https://github.com/NASA-PDS/pds-template-repo-python/pull/44) for the relevant changes. I tested them against the latest in this branch prior to making that PR so it should work nicely unless I really messed something up :)

Reply:
Yeesh … well bringing in the new `..pre-commit-config.yaml` caused all sorts of mayhem, including grammar checks of the docstrings ("First line should end with a period, question mark, or exclamation point") and a hundred+ other errors I'm not about to begin to understand 😱

I'm putting it back to the way it was for now 😬

Reply:
Oh yes, I could see that. The new call isn't running only on files being changed now so it's going to catch a lot more. Good for new code bases, less so for preexisting ones.

There's a way to deal with this in `flake8`. I'll figure it out and open a ticket ...